src/services/bufferUtils.ts

```diff
@@ -13,7 +13,7 @@ export function bufferToString(buffer: Uint8Array): string {
 /**
  * @param str - The string to convert to a buffer (Uint8Array)
  */
-export function stringToBuffer(str: string): Uint8Array {
+export function stringToBuffer(str: string): Uint8Array<ArrayBuffer> {
```

Comment:
FYI: Uint8Array is by default `Uint8Array<ArrayBufferLike>` which does not contain necessarily an `ArrayBuffer`. So `crypto.importKey` would complain.
See also https://devblogs.microsoft.com/typescript/announcing-typescript-5-7/#support-for---target-es2024-and---lib-es2024

(you could e.g. have a shared array buffer which will not work).

So just strong type here as we only use real arrays with array buffer implementation